cocos/3d/lights/point-light-component.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright (c) 2023 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import { ccclass, help, executeInEditMode, menu, tooltip, type, displayOrder,
+    serializable, formerlySerializedAs } from 'cc.decorator';
+import { legacyCC } from '../../core/global-exports';
+import { scene } from '../../render-scene';
+import { Camera } from '../../render-scene/scene';
+import { Light, PhotometricTerm } from './light-component';
+
+/**
+ * @en The point light component, multiple point lights can be added to one scene.
+ * @zh 点光源组件，场景中可以添加多个点光源。
+ */
+@ccclass('cc.PointLight')
+@help('i18n:cc.PointLight')
+@menu('Light/PointLight')
+@executeInEditMode
+export class PointLight extends Light {
+    @serializable
+    @formerlySerializedAs('_luminance')
+    protected _luminanceHDR = 1700 / scene.nt2lm(0.15);
+    @serializable
+    protected _luminanceLDR = 1700 / scene.nt2lm(0.15) * Camera.standardExposureValue * Camera.standardLightMeterScale;
+    @serializable
+    protected _term = PhotometricTerm.LUMINOUS_FLUX;
+    @serializable
+    protected _range = 1;
+
+    protected _type = scene.LightType.POINT;
+    protected _light: scene.PointLight | null = null;
```

Comment:
Why use `protected`? Can it be inherited?

Reply:
it cannot be inherited, but there is a new specification for the use of sphere-light-component, which is used by the original component classes of the engine?

Reply:
If it can not be inherited, it is better to use private. Is there any problem using `private`?

Reply:
will be inconsistent with the scope of the attributes of other components, such as cc.sphere-light-component

Reply:
No, i think we should fix `cc.sphere-light-component` in future too.

Reply:
> 

sure

Reply:
Class 'PointLight' incorrectly extends base class 'Light'.
  Property '_type'  and '_light' is private in type 'PointLight' but not in type 'Light'

I changed all other properties to private

Reply:
> Class 'PointLight' incorrectly extends base class 'Light'.
> Property '_type' and '_light' is private in type 'PointLight' but not in type 'Light'

I don't quite understand the reason why can not use private for `_type` and `_light`?

Reply:
Because point-light-component inherits from the base class light-component, the base class needs to operate on the '_type' and '_light' attributes, and the derived class needs to operate on the '_type' and '_light' attributes, so the base class is defined as 'protected' and then overridden in the derived class The base class is defined as 'protected' and then overridden in the derived class. light-component does not allow the '_type' and '_light' attributes to be exposed

Reply:
Not necessary